### PR TITLE
Allow ignore_last_class w/ External Loss Functions

### PR DESCRIPTION
**Pull request by jamesmcclain**
## Overview

Allow the last class to be ignored when an external loss function is used.  This is done by passing `'force'` to `ignore_last_class` rather than `True` or `False`.

### Checklist

- [x] Updated `docs/changelog.rst`
- [x] Added `needs-backport` label if PR is bug fix that applies to previous minor release
- [x] Ran scripts/format_code and committed any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness

### CODE REVIEW

**Comment by AdeelH** [comment]
If I am understanding this and #1017 correctly, `ignore_last_class=True` does 2 things:
	1. sets `ignore_idx` in `CrossEntropyLoss()`
	2. modifies the model so that it predicts one class less

In case of an external loss function, (1) does not happen, but one might still want (2), though it is not strictly necessary (because if the loss func is ignoring that index anyway, modifying the model does not really have much effect. An unmodified model would not receive any gradient on the weights that predict that last class anyway).

So, I think if all this is clearly noted in the in the documentation for `ignore_last_class`, `'force'` is not needed and just 'True' will suffice.

**Comment by jamesmcclain** [comment]
> If I am understanding this and #1017 correctly, `ignore_last_class=True` does 2 things:
> 1. sets `ignore_idx` in `CrossEntropyLoss()`
> 2. modifies the model so that it predicts one class less
> 
> In case of an external loss function, (1) does not happen, but one might still want (2), though it is not strictly necessary (because if the loss func is ignoring that index anyway, modifying the model does not really have much effect. An unmodified model would not receive any gradient on the weights that predict that last class anyway).
> 
> So, I think if all this is clearly noted in the in the documentation for `ignore_last_class`, `'force'` is not needed and just `True` will suffice.

It is true that the last class does not get any attention from the gradient, but that is part of the problem.  It is clear that its existence still has some effect on the model because I noticed the `null` class continuing to be predicted well into training when starting from a pretrained model.

**Comment by AdeelH** [comment]
Fair enough. I still think the description should be clearer about what the difference between `True` and `'force'` is.

File: rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py [hunk 1]
```diff
@@ -210,7 +211,7 @@ def validate_config(self):
         has_weights = self.class_loss_weights is not None
         has_external_loss_def = self.external_loss_def is not None
 
-        if self.ignore_last_class and has_external_loss_def:
+        if self.ignore_last_class is True and has_external_loss_def:
             raise ConfigError(
                 'ignore_last_class is not supported with external_loss_def.')
```
Comment (AdeelH) [inline rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py:216]: And maybe this error should tell the user to use `'force'` instead

**Review by AdeelH — CHANGES_REQUESTED** [review]

**Comment by jamesmcclain** [comment]
@AdeelH I have addressed your comment

**Review by AdeelH — APPROVED** [review]

**Review by lewfish — APPROVED** [review]
